src/components/inputs/skylarkObjectFieldInput/skylarkObjectFieldInput.component.tsx

```diff
@@ -0,0 +1,234 @@
+import { Select } from "src/components/inputs/select";
+import { INPUT_REGEX, SYSTEM_FIELDS } from "src/constants/skylark";
+import {
+  NormalizedObjectField,
+  SkylarkObjectMetadataField,
+} from "src/interfaces/skylark";
+import { parseInputFieldValue } from "src/lib/skylark/parsers";
+import {
+  formatObjectField,
+  convertFieldTypeToHTMLInputType,
+} from "src/lib/utils";
+
+interface SkylarkObjectFieldInputProps {
+  field: string;
+  config: NormalizedObjectField;
+  register: UseFormRegister<FieldValues>;
+  control: Control<Record<string, SkylarkObjectMetadataField>>;
+  value: SkylarkObjectMetadataField;
+  formState: FormState<Record<string, SkylarkObjectMetadataField>>;
+  additionalRequiredFields?: string[];
+}
+
+interface SkylarkObjectFieldInputComponentProps
+  extends SkylarkObjectFieldInputProps {
+  error?: FieldError;
+  registerOptions?: RegisterOptions<FieldValues, string>;
+}
+
+const createHtmlForId = (field: SkylarkObjectFieldInputProps["field"]) =>
+  `skylark-object-field-input-${field}`;
+
+const SkylarkObjectFieldInputLabel = ({
+  field,
+  isRequired,
+}: {
+  field: SkylarkObjectFieldInputProps["field"];
+  isRequired?: boolean;
+}) => (
+  <label className="mb-2 block font-bold" htmlFor={createHtmlForId(field)}>
+    {formatObjectField(field)}
+    {isRequired && <span className="pl-0.5 text-error">*</span>}
+  </label>
+);
+
+const SkylarkObjectFieldInputEnum = ({
+  field,
+  control,
+  config,
+  error,
+}: SkylarkObjectFieldInputComponentProps) => (
+  <Controller
+    name={field}
+    control={control}
+    render={({ field }) => (
+      <Select
+        className="w-full"
+        variant="primary"
+        selected={(field.value as string) || ""}
+        options={
+          config.enumValues?.map((opt) => ({
+            value: opt,
+            label: opt,
+          })) || []
+        }
+        placeholder={`Select ${formatObjectField(field.name)}`}
+        onChange={field.onChange}
+        aria-invalid={error ? "true" : "false"}
+      />
+    )}
+  />
+);
+
+const SkylarkObjectFieldInputBoolean = ({
+  field,
+  control,
+  error,
+}: SkylarkObjectFieldInputComponentProps) => (
+  <Controller
+    name={field}
+    control={control}
+    render={({ field }) => (
+      <Checkbox
+        checked={field.value as CheckedState}
+        onCheckedChange={field.onChange}
+        aria-invalid={error ? "true" : "false"}
+        id={createHtmlForId(field.name)}
+      />
+    )}
+  />
+);
+
+const SkylarkObjectFieldInputTextArea = ({
+  field,
+  register,
+  value,
+  error,
+  registerOptions,
+}: SkylarkObjectFieldInputComponentProps) => (
+  <textarea
+    {...register(field, registerOptions)}
+    aria-invalid={error ? "true" : "false"}
+    id={createHtmlForId(field)}
+    rows={
+      (value &&
+        (((value as string).length > 1000 && 18) ||
+          ((value as string).length > 150 && 9) ||
+          ((value as string).length > 50 && 5))) ||
+      1
```

Comment:
Do we want a `textarea` with one row? that is an `input`

Reply:
So they can add more content and expand it